proxy-robots-module/src/main/java/orwell/proxy/ProxyRobots.java

```diff
@@ -66,15 +63,14 @@ private void connectToServer() {
      */
     protected void initializeTanksFromConfig() {
         for (final ConfigTank configTank : configRobots.getConfigRobotsToRegister()) {
-            final Camera camera = new Camera(configTank.getConfigCamera().getIp(),
-                    configTank.getConfigCamera().getPort());
-            //TODO Improve initialization of setImage to get something meaningful
-            //from the string (like an actual picture)
-            final Tank tank = new Tank(configTank.getBluetoothName(),
-                    configTank.getBluetoothID(), camera, configTank.getImage());
-            logback.info("Temporary routing ID: " + configTank.getTempRoutingID());
-            tank.setRoutingId(configTank.getTempRoutingID());
-            this.robotsMap.add(tank);
+            final LegoTank tank = RobotFactory.getLegoTank(configTank);
+            if (null == tank) {
+                logback.error("Lego tank not initialized. Skipping it for now.");
```

Comment:
I am not sure this class should know about the implementation of the robots (here it knows it is a Lego tank).